### ISSUE TITLE
DNU tools only work on windows

### ISSUE THREAD

**Issue by andycmaj** [cake-build/cake#689]
[DnuTool.cs](https://github.com/cake-build/cake/blob/develop/src/Cake.Common/Tools/DNU/DNUTool.cs#L46) defines only `DNU.cmd` as the tool executable name.

```csharp
protected override IEnumerable<string> GetToolExecutableNames()
        {
            return new[] { "dnu.cmd" };
        }
```

This doesn't work on unix/osx.
```
Error: DNU: Could not locate executable.
```

I can fake it if i add the `ToolPath` manually in `DNURestoreSettings` in my **build.cake**, but this is not very portable :)

```csharp
Task("Restore")
    .IsDependentOn("Clean")
    .Does(() =>
{
    DNURestore(new DNURestoreSettings { ToolPath = "/Users/acunningham/.dnx/runtimes/dnx-mono.1.0.0-rc1-final/bin/dnu" });
});
```

A possible solution might be to look in `$PATH` for `dnu` executable, since `dnvm` adds the current runtime to the `$PATH`...

```bash
acunningham@ACMBP $ dnvm install 1.0.0-rc1-update1
dnx-mono.1.0.0-rc1-update1 already installed in /Users/acunningham/.dnx
runtime=mono
Adding /Users/acunningham/.dnx/runtimes/dnx-mono.1.0.0-rc1-update1/bin to process PATH
```

**Comment by andycmaj** [comment]
Actually, looking at https://github.com/cake-build/cake/blob/develop/src/Cake.Core/Tooling/Tool.cs#L236, it seems all you'd need to do is add `"dnu"` to `GetToolExecutableNames()`... i'll submit a PR.

**Comment by patriksvensson** [comment]
Merged via #690.
